File: columbu/CATscala/src/main/scala/com/columbu/p2p/udp/message/backup/KeepAliveMessage.scala
package com.galileo.p2p.udp.message.backup

import com.galileo.core.enumeration.UdpMessageTypeEnum
import com.galileo.p2p.udp.message.Message
import com.galileo.p2p.discover.node.Node
import com.galileo.protos.protos.Discover.BackupMessage

class KeepAliveMessage extends Message{

  private var backupMessage:BackupMessage = null

  def this(data: Array[Byte]) {
    this()
    this.`type` = UdpMessageTypeEnum.BACKUP_KEEP_ALIVE
    this.data = data
    this.backupMessage = BackupMessage.parseFrom(data)
  }

  def this(flag: Boolean, priority: Int) {
    this()
    this.`type` = UdpMessageTypeEnum.BACKUP_KEEP_ALIVE
    this.backupMessage = BackupMessage.newBuilder.setFlag(flag).setPriority(priority).build
    this.data = backupMessage.toByteArray
  }

  def getFlag: Boolean = backupMessage.getFlag

  def getPriority: Int = backupMessage.getPriority

  override def getFrom: Node = {
    return null
  }

}

File: columbu/CATscala/src/main/java/com/columbu/p2p/udp/message/backup/KeepAliveMessage.java
package com.galileo.p2p.udp.message.backup;

import static  com.galileo.p2p.udp.message.UdpMessageTypeEnum.BACKUP_KEEP_ALIVE;

import com.galileo.p2p.udp.message.Message;
import com.galileo.core.overlay.discover.node.Node;
import com.galileo.protos.protos.Discover;

public class KeepAliveMessage extends Message {

  private Discover.BackupMessage backupMessage;

  public KeepAliveMessage(byte[] data) throws Exception {
    super(BACKUP_KEEP_ALIVE, data);
    backupMessage = Discover.BackupMessage.parseFrom(data);
  }

  public KeepAliveMessage(boolean flag, int priority) {
    super(BACKUP_KEEP_ALIVE, null);
    backupMessage = Discover.BackupMessage.newBuilder().setFlag(flag).setPriority(priority).build();
    data = backupMessage.toByteArray();
  }

  public boolean getFlag(){
    return backupMessage.getFlag();
  }

  public int getPriority(){
    return backupMessage.getPriority();
  }

  @Override
  public Node getFrom() {
    return null;
  }
}
